samplers/parser.go

```diff
@@ -47,6 +49,42 @@ func (m *MetricKey) String() string {
 	return buff.String()
 }
 
+// ParseMetricSSF converts an incoming SSF packet to a Metric.
+func ParseMetricSSF(metric *ssf.SSFSample) (*UDPMetric, error) {
+	ret := &UDPMetric{
+		SampleRate: 1.0,
+	}
+	h := fnv.New32a()
+	h.Write([]byte(metric.Name))
+	ret.Name = metric.Name
+	switch metric.Metric {
+	case ssf.SSFSample_COUNTER:
+		ret.Type = "counter"
+	case ssf.SSFSample_GAUGE:
+		ret.Type = "gauge"
+	case ssf.SSFSample_HISTOGRAM:
+		ret.Type = "histogram"
+	case ssf.SSFSample_SET:
+		ret.Type = "set"
+	default:
+		return nil, errors.New("Invalid type for metric")
```

Comment:
nit: I have mixed feelings about `errors.New`. Where my feelings are less mixed, though: If the argument to `errors.New` is a constant string literal, we should define that error at the package-level (as an unexported variable), which allows us to check for equality.


```go
var nameOfError = errors.New("Invalid type for metric")

func Foo() error {
    return nameOfError
}
```